drivers/emul/eeprom_at24_emul.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT atmel_at24
+
+#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(atmel_at24);
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <drivers/i2c_emul.h>
+
+#define REG_COUNT	0x100
+
+/** Defines run-time data used by the emulator */
+struct at24_emul_data {
+	/** Size of EEPROM */
+	uint16_t size;
+	/** EEPROM data contents */
+	uint8_t *buf;
+	/** Current register to read */
+	uint8_t cur_reg;
+	/** Current I2C config */
```

Comment:
Why is this here?  Shouldn't it be retrieved from the I2C controller at runtime?

Reply:
At present cur_reg is set when you do a byte write to the device.

So you can do:
- write address
- read bytes
as separate I2C transsaction